addons/account/models/account_analytic_line.py

```diff
@@ -80,6 +80,47 @@ def action_view_vendor_bill(self):
         }
         return result
 
+
+class AccountAnalyticGroup(models.Model):
+    _inherit = 'account.analytic.group'
+
+    applicability_ids = fields.One2many('account.analytic.applicability', 'analytic_group_id', string='Applicability')
+
+    @api.model
+    def get_applicability(self, domain=None, account_id=None, categ_id=None):
+        best_index = 0
+        res = self.default_applicability
+        for rec in self.applicability_ids:
+            index = 0
+            if rec.domain == domain: index += 1
+            if rec.account_code and account_id in self.env['account.account'].search([('code', 'ilike', rec.account_code + '%')]): index +=1
+            if rec.categ_id == categ_id: index += 1
+            if index > best_index:
+                res = rec.applicability
+                best_index = index
+        return res
+
+
+class AccountAnalyticApplicability(models.Model):
+    _name = 'account.analytic.applicability'
+    _description = "Analytic Plan's Applicabilities"
```

Comment:
applicability should probably be defined in the `analytic` module, except for the `account_code` field that would be added in `account`

Reply:
Should we? `product` is not installed before `analytic` either, and I don't know if the domain is useful without `account` (it would be the only thing that would define this applicability). The `default_applicability` is defined in `analytic`. (But the `get_applicability` function on plans should be defined in `analytic` instead).